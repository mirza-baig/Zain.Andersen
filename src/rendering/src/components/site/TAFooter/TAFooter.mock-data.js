const defaultData = {
  rendering: {
    componentName: 'TAFooter',
    dataSource:
      '/sitecore/content/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Home/Components/ProductGallery/Data/TA Footer',
  },
  fields: {
    copyrightText: {
      value: 'Â© {currentYear} Renewal by Andersen LLC. All rights reserved.',
    },
    awLogo: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/RenewalByAndersen/shared/ANDERSEN_Logo_TM_Rectangle_RGB_white.svg?h=37&iar=0&w=260',
        alt: 'ANDERSEN_Logo_TM_Rectangle_RGB_white',
        width: '260',
        height: '37',
      },
    },
    awLogoLink: {
      value: {
        href: '/',
        text: '',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    rbaLogo: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/RenewalByAndersen/shared/LOGO-RbA-Horizontal-white.png?h=455&iar=0&w=1182',
        alt: 'LOGO-RbA-Horizontal-white',
        width: '1182',
        height: '455',
      },
    },
    rbaLogoLink: {
      value: {
        href: '/',
        text: '',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{389C03C7-02A0-463C-B02B-4ED091CAAD15}',
      },
    },
    legalLinks: [
      {
        id: '492ed1cf-51e8-4178-9148-49da1f3695b4',
        url: '/Components/ProductGallery/Data/TA-Footer/Legal-Links/Terms',
        name: 'Terms',
        displayName: 'Terms',
        fields: {
          link: {
            value: {
              href: 'https://www.renewalbyandersen.com/about/terms-of-use',
              linktype: 'external',
              url: 'https://www.renewalbyandersen.com/about/terms-of-use',
            },
          },
          linkTitle: {
            value: 'Terms',
          },
        },
        templateId: '68db4e89-fcf0-4858-adf0-a67945650de0',
        templateName: 'Navigation Link',
      },
      {
        id: '5344db7f-f108-40c4-9f26-526b3d8ed1b7',
        url: '/Components/ProductGallery/Data/TA-Footer/Legal-Links/Privacy-Policy',
        name: 'Privacy Policy',
        displayName: 'Privacy Policy',
        fields: {
          link: {
            value: {
              href: 'https://www.andersenwindows.com/support/privacy/',
              linktype: 'external',
              url: 'https://www.andersenwindows.com/support/privacy/',
            },
          },
          linkTitle: {
            value: 'Privacy Policy',
          },
        },
        templateId: '68db4e89-fcf0-4858-adf0-a67945650de0',
        templateName: 'Navigation Link',
      },
      {
        id: '6e4fa7de-c46d-420d-9c55-c24e0360446f',
        url: '/Components/ProductGallery/Data/TA-Footer/Legal-Links/Privacy-Notice-for-CA-Residents',
        name: 'Privacy Notice for CA Residents',
        displayName: 'Privacy Notice for CA Residents',
        fields: {
          link: {
            value: {
              href: 'https://www.andersenwindows.com/support/privacy-california/',
              linktype: 'external',
              url: 'https://www.andersenwindows.com/support/privacy-california/',
            },
          },
          linkTitle: {
            value: 'Privacy Notice for CA Residents',
          },
        },
        templateId: '68db4e89-fcf0-4858-adf0-a67945650de0',
        templateName: 'Navigation Link',
      },
      {
        id: 'e1d65ceb-f564-40d3-b0db-e75485a883f7',
        url: '/Components/ProductGallery/Data/TA-Footer/Legal-Links/EEO-Policy',
        name: 'EEO Policy',
        displayName: 'EEO Policy',
        fields: {
          link: {
            value: {
              href: '/Components',
              text: '',
              anchor: '',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{7ED4942D-0375-4A10-AE3D-6E7E1361A7EB}',
            },
          },
          linkTitle: {
            value: 'EEO Policy',
          },
        },
        templateId: '68db4e89-fcf0-4858-adf0-a67945650de0',
        templateName: 'Navigation Link',
      },
      {
        id: '492436fd-6708-4557-9217-c031f0db4cfe',
        url: '/Components/ProductGallery/Data/TA-Footer/Legal-Links/Opt-Out-of-Sale-of-Personal-Information',
        name: 'Opt Out of Sale of Personal Information',
        displayName: 'Opt Out of Sale of Personal Information',
        fields: {
          link: {
            value: {
              href: '/Components',
              text: '',
              anchor: '',
              linktype: 'internal',
              class: '',
              title: '',
              target: '',
              querystring: '',
              id: '{7ED4942D-0375-4A10-AE3D-6E7E1361A7EB}',
            },
          },
          linkTitle: {
            value: 'EEO Policy',
          },
        },
        templateId: '68db4e89-fcf0-4858-adf0-a67945650de0',
        templateName: 'Navigation Link',
      },
    ],
    legalText: {
      value:
        'Andersen collects certain categories of personal information. See links for more information.',
    },
    siteSwitcherFlag: {
      value: {
        src: 'https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/RenewalByAndersen/shared/flag_canada.png?h=24&iar=0&w=24',
        alt: 'flag canada',
        width: '24',
        height: '24',
      },
    },
    siteSwitcherLink: {
      value: {
        href: 'https://www.renewalbyandersen.ca/',
        linktype: 'external',
        url: 'https://www.renewalbyandersen.ca/',
      },
    },
    siteSwitcherTitle: {
      value: 'Visit Canadian Site',
    },
    socialLinks: [
      {
        id: '5d901965-46e7-4711-a1fe-3640608dc344',
        url: '/Components/ProductGallery/Data/TA-Footer/Social-Links/LinkedIn',
        name: 'LinkedIn',
        displayName: 'LinkedIn',
        fields: {
          icon: {
            id: '1f1056bf-de5c-4b58-b54a-a0d87db46fb4',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/LinkedIn',
            name: 'LinkedIn',
            displayName: 'LinkedIn',
            fields: {
              Value: {
                value: 'linkedin',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          link: {
            value: {
              href: 'https://www.linkedin.com/company/renewal-by-andersen/ ',
              linktype: 'external',
              url: 'https://www.linkedin.com/company/renewal-by-andersen/ ',
            },
          },
        },
        templateId: '0a3f014a-f0ae-48d9-ac2e-5c78fa7d034c',
        templateName: 'Social Link',
      },
      {
        id: '4651f78e-2770-4e67-bba1-d282fd301e0a',
        url: '/Components/ProductGallery/Data/TA-Footer/Social-Links/Facebook',
        name: 'Facebook',
        displayName: 'Facebook',
        fields: {
          icon: {
            id: 'c7f9916c-dcc5-4439-a50e-103ad81854b1',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Facebook',
            name: 'Facebook',
            displayName: 'Facebook',
            fields: {
              Value: {
                value: 'facebook',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          link: {
            value: {
              href: 'https://www.facebook.com/rbacorp',
              linktype: 'external',
              url: 'https://www.facebook.com/rbacorp',
            },
          },
        },
        templateId: '0a3f014a-f0ae-48d9-ac2e-5c78fa7d034c',
        templateName: 'Social Link',
      },
      {
        id: 'b671d185-8f02-44d1-b010-57986f917a56',
        url: '/Components/ProductGallery/Data/TA-Footer/Social-Links/Twitter',
        name: 'Twitter',
        displayName: 'Twitter',
        fields: {
          icon: {
            id: '8c7cc7f8-8290-4546-8d0f-2d8ed1a82941',
            url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Twitter',
            name: 'Twitter',
            displayName: 'Twitter',
            fields: {
              Value: {
                value: 'twitter',
              },
            },
            templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
            templateName: 'Enum',
          },
          link: {
            value: {
              href: 'https://twitter.com/RbACorp',
              linktype: 'external',
              url: 'https://twitter.com/RbACorp',
            },
          },
        },
        templateId: '0a3f014a-f0ae-48d9-ac2e-5c78fa7d034c',
        templateName: 'Social Link',
      },
    ],
    socialLinksText: {
      value: 'Follow us on',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
