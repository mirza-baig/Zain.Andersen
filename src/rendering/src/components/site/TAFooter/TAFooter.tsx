// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import {
  Field,
  ImageField,
  LinkField,
  Text,
  withDatasourceCheck,
} from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { ComponentProps } from 'lib/types/component-props';
import { LinkWrapper } from 'src/helpers/LinkWrapper';
import Image from 'next/image';
import classNames from 'classnames';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { getEnum, useExperienceEditor } from 'lib/utils';
import { IconTypes, SvgIcon } from 'src/helpers/SvgIcon';
import { Subheadline } from 'src/helpers/Subheadline';
import { ReactElement, useEffect, useState } from 'react';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';
import { isSvgUrl } from 'lib/utils/string-utils';

export type TAFooterProps =
  Feature.EnterpriseWeb.RenewalByAndersen.Components.Navigation.TAFooter & {
    fields?: {
      socialLinks: Feature.EnterpriseWeb.Enterprise.Elements.Social.SocialLink[];
    };
  } & ComponentProps;

const TAFooter = (props: TAFooterProps) => {
  const isEE = useExperienceEditor();

  const [siteSwitcherLink, setSiteSwticherLink] = useState<null | LinkField>(null);

  const getSiteSwitcherLink = (): LinkField | null => {
    // Check if running on the client side
    if (typeof window !== 'undefined') {
      // Determine the current domain and switch it
      const switchedUrl: string = window.location.href.replace(
        /\.(com|ca)/g,
        (_: string, domain: string) => (domain === 'com' ? '.ca' : '.com')
      );

      const linkField = {
        value: {
          href: switchedUrl,
          linktype: 'external',
          url: switchedUrl,
        },
      };

      return linkField;
    }

    return null;
  };

  useEffect(() => {
    setSiteSwticherLink(getSiteSwitcherLink());
  }, []);

  if (!props.fields) {
    return <></>;
  }

  const currentYear = new Date().getFullYear().toString();
  let textWithReplacedYear;
  if (props.fields.copyrightText.value?.includes('{currentYear}')) {
    textWithReplacedYear = props.fields.copyrightText.value.replace('{currentYear}', currentYear);
  }

  const rbaLogo = {
    value: {
      ...props.fields.rbaLogo.value,
      width: '143',
      height: '49',
    },
  };

  const awLogo = {
    value: {
      ...props.fields.awLogo.value,
      width: '188',
      height: '26',
    },
  };

  return (
    <Component
      variant="full"
      padding="px-m py-m ml:pt-ml ml:pb-m"
      backgroundVariant="black"
      dataComponent="site/tafooter"
      {...props}
    >
      <div className="col-span-12 mx-auto grid w-full max-w-[1200px] grid-cols-2 gap-s ml:grid-cols-12 ml:gap-y-xxxs ">
        {/* Site Logo Section */}
        <div className="col-span-2  text-white ml:col-span-8">
          <div className="flex flex-row gap-x-xxs ml:gap-x-s">
            {props.fields?.rbaLogo && (
              <FooterLogo logoImg={rbaLogo} logoLink={props.fields.rbaLogoLink} />
            )}
            {props.fields?.awLogo && (
              <FooterLogo logoImg={awLogo} logoLink={props.fields.awLogoLink} />
            )}
          </div>
        </div>
        {/* Social Icons Section */}
        <div className="col-span-2 flex flex-col flex-wrap justify-end gap-x-s  text-white ml:col-span-4 ml:flex-row ml:items-center">
          <BodyCopy
            fields={{ body: props.fields.socialLinksText || { value: '' } }}
            classes="mb-0 ml:text-right font-bold text-body"
          />
          <ul className="flex gap-x-xxs">
            {props.fields.socialLinks?.map(
              (
                socialLink: Feature.EnterpriseWeb.Enterprise.Elements.Social.SocialLink,
                index: number
              ) => {
                const icon = getEnum<IconTypes>(socialLink.fields.icon);
                return (
                  icon &&
                  socialLink?.fields?.link?.value && (
                    <li key={index}>
                      {isEE ? (
                        <SvgIcon icon={icon} />
                      ) : (
                        <LinkWrapper
                          ctaSection="footer"
                          field={socialLink.fields.link}
                          ariaLabel={{
                            value: socialLink.fields.icon?.displayName || 'Social Icon',
                          }}
                        >
                          <SvgIcon icon={icon} />
                        </LinkWrapper>
                      )}
                    </li>
                  )
                );
              }
            )}
          </ul>
        </div>
        {/* Legal Links Section */}
        <div className="col-span-2  mb-xxs text-white ml:col-span-8 ml:mb-0">
          <div className="flex flex-col">
            <Subheadline
              fields={{ subheadlineText: props.fields.legalText }}
              classes="mb-m ml:mb-xxxs text-small font-bold !font-serif"
            />
            <ul>
              {props.fields.legalLinks?.map((legalLink: Item, index: number) => {
                return (
                  legalLink.fields?.link && (
                    <li
                      key={index}
                      className={classNames(
                        'relative inline px-s text-small',
                        index != 0
                          ? 'before:absolute before:top-[3px] before:left-0 before:inline-block before:h-[11px] before:border-l '
                          : 'pl-0'
                      )}
                    >
                      <LinkWrapper
                        ctaSection="footer"
                        className="underline-offset-1 hover:underline"
                        field={legalLink.fields.link as LinkField}
                        suppressLinkText
                        ariaLabel={{
                          value: legalLink?.displayName || 'Legal Link',
                        }}
                      >
                        <Text
                          tag=""
                          field={(legalLink.fields.linkTitle as Field<string>) || { value: '' }}
                        />
                      </LinkWrapper>
                    </li>
                  )
                );
              })}
            </ul>
          </div>
        </div>
        {/* Site Switcher Section */}
        <div className="col-span-2  text-white ml:col-span-4">
          {siteSwitcherLink && (
            <LinkWrapper
              field={siteSwitcherLink}
              ctaSection="footer"
              className="mb-m flex w-fit items-center gap-x-xs  hover:underline ml:mb-xxxs ml:ml-auto ml:justify-end"
              ariaLabel={{
                value: props.fields.siteSwitcherFlag.value?.alt as string,
              }}
            >
              {props.fields.siteSwitcherFlag.value?.src && (
                <Image
                  src={props.fields.siteSwitcherFlag.value?.src}
                  alt={(props.fields.siteSwitcherFlag.value?.alt as string) || ''}
                  height={props.fields.siteSwitcherFlag.value?.height as string}
                  width={props.fields.siteSwitcherFlag.value?.width as string}
                  layout="intrinsic"
                  unoptimized={isSvgUrl(props.fields.siteSwitcherFlag.value?.src)}
                />
              )}
              <Subheadline
                useTag="span"
                fields={{ subheadlineText: props.fields.siteSwitcherTitle }}
                classes="text-body font-bold !font-serif"
              />
            </LinkWrapper>
          )}

          <div className="text-sm-xxs font-regular ml:text-right">
            <Text field={{ value: textWithReplacedYear }} />
          </div>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<TAFooterProps>(TAFooter);

interface FooterLogoProps {
  logoImg: ImageField;
  logoLink: LinkField;
  classes?: string;
}

const FooterLogo = (props: FooterLogoProps): ReactElement => {
  if (props.logoImg?.value?.src) {
    return (
      <LinkWrapper
        ctaSection="footer"
        field={props.logoLink}
        className={classNames('flex h-fit justify-center', props.classes)}
        ariaLabel={{
          value: props.logoImg.value?.alt as string,
        }}
      >
        <Image
          src={props.logoImg.value?.src}
          alt={(props.logoImg.value?.alt as string) || ''}
          height={props.logoImg.value?.height as string}
          width={props.logoImg.value?.width as string}
          layout="intrinsic"
        />
      </LinkWrapper>
    );
  }
  return <></>;
};
