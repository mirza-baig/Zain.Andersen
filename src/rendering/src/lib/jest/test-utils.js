// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
