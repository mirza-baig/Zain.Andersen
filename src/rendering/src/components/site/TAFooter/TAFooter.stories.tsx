// Global
import { Story, Meta } from '@storybook/react';
// Local
import TAFooter, { TAFooterProps } from './TAFooter';
import defaultData from './TAFooter.mock-data';

export default {
  title: 'Site/TAFooter',
  component: TAFooter,
} as Meta;
const Template: Story<TAFooterProps> = (props) => <TAFooter {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
