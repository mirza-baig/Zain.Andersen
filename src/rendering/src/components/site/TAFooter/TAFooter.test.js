// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import TAFooter from './TAFooter';
import defaultData from './TAFooter.mock-data';

it('renders correctly', async () => {
  const component = snapshot(TAFooter, { componentProps: defaultData });
  await hasDataComponent(component, 'site/tafooter');
});
