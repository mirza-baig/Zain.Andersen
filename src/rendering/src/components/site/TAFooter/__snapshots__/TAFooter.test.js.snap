// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="site/tafooter"
  id="id343697139"
>
  <div
    class="my-8 w-full theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m py-m ml:pt-ml ml:pb-m gap-s md:gap-s "
    >
      <div
        class="col-span-12 mx-auto grid w-full max-w-[1200px] grid-cols-2 gap-s ml:grid-cols-12 ml:gap-y-xxxs "
      >
        <div
          class="col-span-2 text-white ml:col-span-8"
        >
          <div
            class="flex flex-row gap-x-xxs ml:gap-x-s"
          >
            <a
              aria-label="LOGO-RbA-Horizontal-white"
              class="flex h-fit justify-center"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27143%27%20height=%2749%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="LOGO-RbA-Horizontal-white"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
            <a
              aria-label="ANDERSEN_Logo_TM_Rectangle_RGB_white"
              class="flex h-fit justify-center"
              data-gtm-click=""
              data-gtm-dl-event="nav_click"
              data-gtm-dl-nav-section="footer"
              href="/"
              role="link"
              tabindex="0"
              target=""
              title=""
            >
              
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27188%27%20height=%2726%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="ANDERSEN_Logo_TM_Rectangle_RGB_white"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </div>
        </div>
        <div
          class="col-span-2 flex flex-col flex-wrap justify-end gap-x-s text-white ml:col-span-4 ml:flex-row ml:items-center"
        >
          <div
            class="mb-0 ml:text-right font-bold text-body"
          >
            <div
              class="body-copy"
            >
              <div>
                Follow us on
              </div>
            </div>
          </div>
          <ul
            class="flex gap-x-xxs"
          >
            <li>
              <a
                aria-label="LinkedIn"
                data-gtm-click=""
                data-gtm-dl-event="nav_click"
                data-gtm-dl-nav-section="footer"
                href="https://www.linkedin.com/company/renewal-by-andersen/ "
                role="link"
                tabindex="0"
              >
                <span />
              </a>
            </li>
            <li>
              <a
                aria-label="Facebook"
                data-gtm-click=""
                data-gtm-dl-event="nav_click"
                data-gtm-dl-nav-section="footer"
                href="https://www.facebook.com/rbacorp"
                role="link"
                tabindex="0"
              >
                <span />
              </a>
            </li>
            <li>
              <a
                aria-label="Twitter"
                data-gtm-click=""
                data-gtm-dl-event="nav_click"
                data-gtm-dl-nav-section="footer"
                href="https://twitter.com/RbACorp"
                role="link"
                tabindex="0"
              >
                <span />
              </a>
            </li>
          </ul>
        </div>
        <div
          class="col-span-2 mb-xxs text-white ml:col-span-8 ml:mb-0"
        >
          <div
            class="flex flex-col"
          >
            <div
              class="font-sans mb-m ml:mb-xxxs text-small font-bold !font-serif"
            >
              <h3>
                Andersen collects certain categories of personal information. See links for more information.
              </h3>
            </div>
            <ul>
              <li
                class="relative inline px-s text-small pl-0"
              >
                <a
                  aria-label="Terms"
                  class="underline-offset-1 hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="https://www.renewalbyandersen.com/about/terms-of-use"
                  role="link"
                  tabindex="0"
                >
                  
                  Terms
                </a>
              </li>
              <li
                class="relative inline px-s text-small before:absolute before:top-[3px] before:left-0 before:inline-block before:h-[11px] before:border-l "
              >
                <a
                  aria-label="Privacy Policy"
                  class="underline-offset-1 hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="https://www.andersenwindows.com/support/privacy/"
                  role="link"
                  tabindex="0"
                >
                  
                  Privacy Policy
                </a>
              </li>
              <li
                class="relative inline px-s text-small before:absolute before:top-[3px] before:left-0 before:inline-block before:h-[11px] before:border-l "
              >
                <a
                  aria-label="Privacy Notice for CA Residents"
                  class="underline-offset-1 hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="https://www.andersenwindows.com/support/privacy-california/"
                  role="link"
                  tabindex="0"
                >
                  
                  Privacy Notice for CA Residents
                </a>
              </li>
              <li
                class="relative inline px-s text-small before:absolute before:top-[3px] before:left-0 before:inline-block before:h-[11px] before:border-l "
              >
                <a
                  aria-label="EEO Policy"
                  class="underline-offset-1 hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/Components/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  
                  EEO Policy
                </a>
              </li>
              <li
                class="relative inline px-s text-small before:absolute before:top-[3px] before:left-0 before:inline-block before:h-[11px] before:border-l "
              >
                <a
                  aria-label="Opt Out of Sale of Personal Information"
                  class="underline-offset-1 hover:underline"
                  data-gtm-click=""
                  data-gtm-dl-event="nav_click"
                  data-gtm-dl-nav-section="footer"
                  href="/Components"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  
                  EEO Policy
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-span-2 text-white ml:col-span-4"
        >
          <a
            aria-label="flag canada"
            class="mb-m flex w-fit items-center gap-x-xs hover:underline ml:mb-xxxs ml:ml-auto ml:justify-end"
            data-gtm-click=""
            data-gtm-dl-event="nav_click"
            data-gtm-dl-nav-section="footer"
            href="http://localhost/"
            role="link"
            tabindex="0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="flag canada"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="font-sans text-body font-bold !font-serif"
            >
              <span>
                Visit Canadian Site
              </span>
            </div>
          </a>
          <div
            class="text-sm-xxs font-regular ml:text-right"
          >
            © 2024 Renewal by Andersen LLC. All rights reserved.
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
